wagtail/core/query.py

```diff
@@ -418,6 +420,53 @@ def not_translation_of(self, page, inclusive=False):
         """
         return self.exclude(self.translation_of_q(page, inclusive))
 
+    def with_workflow_state(self):
```

Comment:
Since these methods do slightly different things, could we replace ``with_`` with something more descriptive?

For example, ``prefetch_workflow_states()``, ``annotate_approved_schedule``, etc?

Reply:
Thanks Karl - that makes sense. I've renamed them and moved them to the appropriate view file.